sql/6_1_0-to-7_0_0_upgrade.sql

```diff
@@ -727,6 +727,67 @@ UPDATE `layout_options` SET `title` = 'Title' WHERE `layout_options`.`form_id` =
 UPDATE `layout_options` SET `title` = 'Name', `titlecols` = '1', `datacols` = '3' WHERE `layout_options`.`form_id` = 'DEM' AND `layout_options`.`field_id` = 'fname';
 #EndIf
 
+#IfMissingColumn addresses district
+ALTER TABLE `addresses` ADD COLUMN `district` VARCHAR(255) DEFAULT NULL COMMENT 'The county or district of the address';
+#EndIf
+
+
+#IfNotTable contact
+CREATE TABLE `contact` (
+   `id` BIGINT(20) NOT NULL auto_increment,
+   `foreign_table_name` VARCHAR(255) NOT NULL DEFAULT '',
+   `foreign_id` BIGINT(20) NOT NULL DEFAULT '0',
+   PRIMARY KEY (`id`),
+   KEY (`foreign_id`)
+) ENGINE = InnoDB;
+#EndIf
+
+
+
+#IfNotTable contact_address
+CREATE TABLE `contact_address` (
+    `id` BIGINT(20) NOT NULL auto_increment,
+    `contact_id` BIGINT(20) NOT NULL,
+    `address_id` BIGINT(20) NOT NULL,
+    `priority` INT(11) NULL,
+    `type` VARCHAR(255) NULL COMMENT 'FK to list_options.option_id for list_id address-types',
+    `use` VARCHAR(255) NULL COMMENT 'FK to list_options.option_id for list_id address-uses',
+    `notes` TEXT(255),
```

Comment:
also need to remove the (255)

Reply:
Is there any particular reason to use TEXT for `notes`? db engines tend to treat TEXT/blobs with storage related handling overheads. Is varchar's max size(21,844 utf8, 65,532 otherwise) not enough? This table should not have max rowsize issues.

Reply:
should use tinytext.

Reply:
I don't particularly care one way or the other.  I will say that in most my designs I typically choose text columns once my column size gets over 256-512 characters.  MySQL would often inline a varchar into the row when joining column data if its under a certain size. Been a while since I've benchmarked this though.   (I just looked it up and they've bumped varchar inlining up to 768 bytes). 

I stuck with text as that was @psoas original design.  I don't know the data requirement on an address note, though 1024 characters is probably sufficient for this if we want it to stay as a varchar.  

Otherwise, I'm not sure what overhead you are referring to in treating text/blobs outside of the disk fetch when the column is included in the select.  But a varchar will have the same problem if it exceeds the inlining in the db engine.

Reply:
Changed this to be TinyText